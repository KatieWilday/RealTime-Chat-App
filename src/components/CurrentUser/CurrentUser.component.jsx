import React from "react";

import { signOut } from "../../firebase/firebase.utils";
import { Link } from "react-router-dom";

import "./CurrentUser.styles.scss";
const CurrentUser = ({ displayName, photoURL, email, children }) => {
  return (
    <section className="CurrentUser">
      <div className="CurrentUser--profile">
        {/* {photoURL && <img src={photoURL} alt={displayName} />} */}
        <div className="CurrentUser--information">
          <h2 className="userDisplay">{displayName}</h2>
          <p className="email">{email}</p>
        </div>
      </div>
      <div>
        <div>{children}</div>
        <Link to="/" onClick={signOut}>
          Sign Out
        </Link>
      </div>
    </section>
  );
};
CurrentUser.defaultProps = {
  displayName: "Bill Murray",
  email: "billmurray@mailinator.com",
  photoURL: "https://www.fillmurray.com/300/300",
  createdAt: new Date()
};
export default CurrentUser;
