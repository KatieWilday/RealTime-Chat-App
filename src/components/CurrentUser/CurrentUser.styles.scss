$text-white: #fff;

.email {
  color: $text-white;
}

.userDisplay {
  color: $text-white;
}
