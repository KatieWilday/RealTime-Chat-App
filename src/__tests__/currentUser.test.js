import React from "react";
import { create } from "react-test-renderer";
import CurrentUser from "../components/CurrentUser/CurrentUser.component";
import { BrowserRouter } from "react-router-dom";

test("snapshot", () => {
  const tree = create(
    <BrowserRouter>
      <CurrentUser />
    </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
