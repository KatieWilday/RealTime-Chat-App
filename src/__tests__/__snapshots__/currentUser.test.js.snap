// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<section
  className="CurrentUser"
>
  <div
    className="CurrentUser--profile"
  >
    <div
      className="CurrentUser--information"
    >
      <h2
        className="userDisplay"
      >
        Bill Murray
      </h2>
      <p
        className="email"
      >
        billmurray@mailinator.com
      </p>
    </div>
  </div>
  <div>
    <div />
    <a
      href="/"
      onClick={[Function]}
    >
      Sign Out
    </a>
  </div>
</section>
`;
